xarray/backends/api.py

```diff
@@ -561,6 +571,9 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
     parallel : bool, optional
         If True, the open and preprocess steps of this function will be
         performed in parallel using ``dask.delayed``. Default is False.
+    combine : {'manual', 'auto'}, optional
+        Whether ``xarray.auto_combine`` or ``xarray.manual_combine`` is used to
+        combine all the data. Default is 'manual'.
```

Comment:
I think it probably makes more sense for the default to be `'auto'`, unless you explicitly provide a list of dimensions to stack along?

Reply:
I agree the default should be `'auto'`, the reason it's not here is because no-one has written the automatic-ordering logic yet.

That should be relatively simple though right? Something like:

```python
def _infer_concat_order_from_coords(datasets, concat_dims):
  
    # Deduce order along each dimension
    tile_ids = [(,) for ds in datasets]
    for dim in concat_dims:
        # Requires reading the dims and values of coordinate variables from all datasets
        unsorted_first_coord_values = np.vstack(ds[dim].values[0] for ds in datasets)

        # Sort along dim 
        # (Presumably use numpy.argsort, but might need special cases for string coords)
        order = np.argsort(unsort_first_coord_values)

        tile_ids = [tile_id + order[i] for i, tile_id in enumerate(tile_ids)]

    combined_ids = OrderedDict(zip(tile_ids, datasets))

    return combined_ids, concat_dims
```

Reply:
I've just realised the approach here doesn't do any auto-merging, only auto-concatenating... Maybe I need to think about this a bit more...

Reply:
This could probably be done in a two step process, like what we currently do:
- Group datasets by their unique data variables. Within each group, concatenate.
- Merge together the results.

Reply:
> This could probably be done in a two step process

So I think you would need to work out the ordering separately for each group of datasets, in case some of them don't contain the same dimensions, but that's okay, i.e:

```python
# Group by data vars
grouped = itertools.groupby(datasets, key=lambda ds: tuple(sorted(ds)))

concatenated = []
for var_group in grouped:
    combined_ids, concat_dims = _infer_order_from_coords(list(var_group))

    concatenated.append(_combine_nd(combined_ids, concat_dims))
    
return merge(concatenated, compat=compat)
    
```

> The alternative would be to check pandas.Index.is_monotonic to verify monotonicity on each individual dataset, and then verify that the last coordinate value of each dataset is less than the first coordinate value of the next dataset.

That's neater and clearer, I'll do that.

Reply:
> That's neater and clearer, I'll try that. Presumably the `pandas.Index` object already exists for each dimension on each dataset, is there a way to access this via some hidden part of the xarray API?

You can find indexes in `Dataset.indexes`